packages/types/src/package.ts

```diff
@@ -22,6 +22,103 @@ export interface Package {
    * The state exposed by this package.
    */
   state?: {
+    /**
+     * The configuration objects for all the fills exposed by this package.
+     */
+    fills?: {
+      [namespace: string]: {
+        [name: string]: Fill;
+      };
+    };
+
+    /**
+     * The rest of the state exposed by this package.
+     *
+     * TODO: Replace `any` with serializable objects and derived state.
+     */
+    [namespace: string]: {
+      [key: string]: any;
+    };
+  };
+
+  /**
+   * The actions exposed by this package.
+   */
+  actions?: {
+    [namespace: string]: Actions;
+  };
+
+  /**
+   * The libraries exposed by this package.
+   */
+  libraries?: {
+    /**
+     * The Fill components exposed by this package.
+     */
+    fills?: {
+      [namespace: string]: {
+        [name: string]: React.ElementType;
+      };
+    };
+
+    /**
+     * The rest of the libraries exposed by this package.
+     */
+    [namespace: string]: {
+      [library: string]: any;
+    };
+  };
+}
+
+export default Package;
+
+/**
+ * An object containing Frontity actions, or other objects containing actions.
+ */
+interface Actions {
+  [key: string]: (...args: any) => any | Actions;
+}
+
+/**
+ * The configuration object for a Fill.
+ */
+export interface Fill {
```

Comment:
Do you think it would be clearer to create individual files for `Frontity`, `Package` and `Fills` instead of exporting them all from `Packages`? I feel like they would be easier to find this way?

Reply:
Totally agree. I'll change that.